bitnami/grafana-alloy/templates/application.yaml

```diff
@@ -0,0 +1,343 @@
+            periodSeconds: {{ .Values.alloy.startupProbe.periodSeconds }}
+            timeoutSeconds: {{ .Values.alloy.startupProbe.timeoutSeconds }}
+            successThreshold: {{ .Values.alloy.startupProbe.successThreshold }}
+            failureThreshold: {{ .Values.alloy.startupProbe.failureThreshold }}
+          {{- end }}
+          {{- if .Values.alloy.lifecycleHooks }}
+          lifecycle: {{- include "common.tplvalues.render" (dict "value" .Values.alloy.lifecycleHooks "context" $) | nindent 12 }}
+          {{- end }}
+          {{- end }}
+          volumeMounts:
+            - name: config
+              mountPath: /opt/bitnami/grafana-alloy/config
+            {{- if .Values.alloy.mounts.varlog }}
+            - name: varlog
+              mountPath: /var/log
+              readOnly: true
+            {{- end }}
+            {{- if .Values.alloy.mounts.dockercontainers }}
+            - name: dockercontainers
+              mountPath: /var/lib/docker/containers
+              readOnly: true
+            {{- end }}
+            - name: empty-dir
+              mountPath: {{ .Values.alloy.storagePath }}
+              subPath: app-storage-dir
+            - name: empty-dir
+              mountPath: /tmp
+              subPath: tmp-dir
+            {{- if .Values.alloy.extraVolumeMounts }}
+            {{- include "common.tplvalues.render" (dict "value" .Values.alloy.extraVolumeMounts "context" $) | nindent 12 }}
+            {{- end }}
+        {{- if .Values.configReloader.enabled }}
+        - name: config-reloader
+          image: {{ template "config-reloader.image" . }}
+          imagePullPolicy: {{ .Values.configReloader.image.pullPolicy }}
+          {{- if .Values.configReloader.containerSecurityContext.enabled }}
+          securityContext: {{- include "common.compatibility.renderSecurityContext" (dict "secContext" .Values.configReloader.containerSecurityContext "context" $) | nindent 12 }}
+          {{- end }}
+          {{- if .Values.diagnosticMode.enabled }}
+          command: {{- include "common.tplvalues.render" (dict "value" .Values.diagnosticMode.command "context" $) | nindent 12 }}
+          {{- else if .Values.configReloader.command }}
+          command: {{- include "common.tplvalues.render" (dict "value" .Values.configReloader.command "context" $) | nindent 12 }}
+          {{- end }}
+          {{- if .Values.diagnosticMode.enabled }}
+          args: {{- include "common.tplvalues.render" (dict "value" .Values.diagnosticMode.args "context" $) | nindent 12 }}
+          {{- else if .Values.configReloader.args }}
+          args: {{- include "common.tplvalues.render" (dict "value" .Values.configReloader.args "context" $) | nindent 12 }}
+          {{- else }}
+          args:
+            - -volume-dir=/opt/bitnami/grafana-alloy/config
+            - -webhook-url=http://localhost:{{ .Values.alloy.containerPorts.http }}/-/reload
```

Comment:
configmapReload has a port, we should configure it. And you can use it in the probes

https://github.com/jimmidyson/configmap-reload/blob/9d735fc1416853c8cdb3e0f8c6c2aadac762a460/configmap-reload.go#L27C1-L27C121